src/kbmod/reprojection.py

```diff
@@ -289,38 +387,52 @@ def _reproject_work_unit_in_parallel(
 
     # when all the multiprocessing has finished, convert the returned numpy arrays to RawImages.
     concurrent.futures.wait(future_reprojections, return_when=concurrent.futures.ALL_COMPLETED)
-    stack = ImageStack([])
-    for result in future_reprojections:
-        science_add, variance_add, mask_add, time = result.result()
-        psf = _get_first_psf_at_time(work_unit, time)
-
-        # And then stack the RawImages into a LayeredImage.
-        new_layered_image = LayeredImage(
-            science_add,
-            variance_add,
-            mask_add,
-            psf,
-            time,
-        )
-        stack.append_image(new_layered_image, force_move=True)
 
-    # sort by the time_stamp
-    stack.sort_by_time()
+    if write_output:
+        for result in future_reprojections:
+            if not result.result():
+                raise RuntimeError("one or more jobs failed.")
 
-    # Add the imageStack to a new WorkUnit and return it.
-    new_wunit = WorkUnit(
-        im_stack=stack,
-        config=work_unit.config,
-        wcs=common_wcs,
-        constituent_images=work_unit.constituent_images,
-        per_image_wcs=work_unit._per_image_wcs,
-        per_image_ebd_wcs=work_unit._per_image_ebd_wcs,
-        per_image_indices=unique_obstimes_indices,
-        geocentric_distances=work_unit.geocentric_distances,
-        reprojected=True,
-    )
+        new_work_unit = copy(work_unit)
+        new_work_unit._per_image_indices = unique_obstimes_indices
+        new_work_unit.wcs = common_wcs
+        new_work_unit.reprojected = True
+
+        hdul = new_work_unit.metadata_to_primary_header()
+        hdul.writeto(os.path.join(directory, filename))
+    else:
+        stack = ImageStack([])
+        for result in future_reprojections:
+            science_add, variance_add, mask_add, time = result.result()
+            psf = _get_first_psf_at_time(work_unit, obstime)
+
+            # And then stack the RawImages into a LayeredImage.
+            new_layered_image = LayeredImage(
+                science_add,
+                variance_add,
+                mask_add,
+                psf,
+                time,
+            )
+            stack.append_image(new_layered_image, force_move=True)
+
+        # sort by the time_stamp
+        stack.sort_by_time()
+
+        # Add the imageStack to a new WorkUnit and return it.
+        new_wunit = WorkUnit(
+            im_stack=stack,
+            config=work_unit.config,
+            wcs=common_wcs,
+            constituent_images=work_unit.constituent_images,
+            per_image_wcs=work_unit._per_image_wcs,
+            per_image_ebd_wcs=work_unit._per_image_ebd_wcs,
+            per_image_indices=unique_obstimes_indices,
+            geocentric_distances=work_unit.geocentric_distances,
+            reprojected=True,
+        )
 
-    return new_wunit
+        return new_wunit
```

Comment:
It looks like you are no longer returning a workunit down this branch. Is that correct? If so, you should update the doc string to note that the function will return None if write_output is set to True.